client/src/js/samples/components/Create/__tests__/Create.test.js

```diff
@@ -0,0 +1,222 @@
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper.find("Modal").simulate("exit");
+        expect(wrapper.state()).toEqual({ ...state, subtraction: "foo" });
+    });
+    it("handleChange() should update state [name] and [error] when InputError is changed and [name=name]", () => {
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper
+            .find("InputError")
+            .at(0)
+            .simulate("change", e);
+        expect(wrapper.state()).toEqual({ ...state, name: "foo" });
+    });
+    it("handleChange() should update [name] when [name!=name] and [name!=subtraction]", () => {
+        e.target.name = "Foo";
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper
+            .find("InputError")
+            .at(0)
+            .simulate("change", e);
+        expect(wrapper.state()).toEqual({ ...state, Foo: "foo" });
+    });
+    it("handleLibrarySelect() should update libraryType when LibraryTypeSelection is selected", () => {
+        const libraryType = "sRNA";
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper
+            .find("LibraryTypeSelection")
+            .at(0)
+            .simulate("select", libraryType);
+        expect(wrapper.state()).toEqual({ ...state, libraryType });
+    });
+
+    it("handleSubmit() should update errorName when form is submitted and [this.state.name='']", () => {
+        props.subtractions = ["foo"];
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper.setState({ ...state, selected: ["bar"] });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({ ...state, selected: ["bar"], errorName: "Required Field" });
+    });
+    it("handleSubmit() should update errorSubtraction when form is submitted and [this.props.subtractions=[]]", () => {
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper.setState({ ...state, name: "foo", selected: ["bar"] });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            errorSubtraction: "At least one subtraction must be added to Virtool before samples can be analyzed.",
+            selected: ["bar"],
+            name: "foo"
+        });
+    });
+    it("handleSubmit() should update errorFile when form is submitted and [this.props.selected=[]]", () => {
+        props.subtractions = ["foo"];
+        const wrapper = shallow(<CreateSample {...props} />);
+        wrapper.setState({ ...state, name: "foo" });
+        wrapper.find("form").simulate("submit", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            errorFile: "At least one read file must be attached to the sample",
+            name: "foo"
+        });
+    });
+    it("handleSubmit() should call onCreate when form is submitted and [hasError=false]", () => {
+        const hasError = false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['hasError' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/virtool/virtool/pullRequest?prid=4685553)